src/BenchmarkDotNet/Loggers/LinqPadLogger.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace BenchmarkDotNet.Loggers
+{
+    public sealed class LinqPadLogger : ILogger
+    {
+        private const string DefaultColor = "";
+
+        private static Type linqPadUtil;
+
+        private static MethodInfo withStyle;
+
+        private readonly IReadOnlyDictionary<LogKind, string> colorScheme;
+
+        private static readonly Lazy<bool> isAvailable = new Lazy<bool>(() =>
+        {
+            if (AppDomain.CurrentDomain.FriendlyName.StartsWith("LINQPad Query Server", StringComparison.OrdinalIgnoreCase))
+            {
+                try
+                {
+                    // Use Reflection to avoid taking a dependency on the LINQPad assembly
+                    linqPadUtil = Type.GetType("LINQPad.Util, LINQPad, Version=1.0.0.0, Culture=neutral, PublicKeyToken=21353812cd2a2db5", throwOnError: true);
+                    withStyle = linqPadUtil.GetMethod("WithStyle", BindingFlags.Static | BindingFlags.Public);

```

Comment:
I'm not happy with a Lazy value provider having the side-effect of setting static variables; will rework.

Reply:
Implemented this change (using static field initialisation for laziness).